File: lib/domain/CoffeeUseCaseImpl.dart
import 'package:technical_test/domain/CoffeeRepository.dart';
import 'package:technical_test/domain/CoffeeUseCase.dart';
import 'package:technical_test/domain/model/CoffeeDomainModel.dart';
import 'package:technical_test/utils/RequestState.dart';

class CoffeeUseCaseImpl implements CoffeeUseCase {
  final CoffeeRepository repository;

  CoffeeUseCaseImpl(this.repository);

  @override
  Future<RequestState<List<CoffeeDomainModel>>> getAllCoffeeHot() {
    return repository.getAllCoffeeHot();
  }

  @override
  Future<RequestState<List<CoffeeDomainModel>>> getAllCoffeeIce() {
    return repository.getAllCoffeeIce();
  }
}

File: app/src/main/java/com/example/coffeepedia/domain/CoffeeUseCaseImpl.kt
package com.example.coffeepedia.domain

import com.example.coffeepedia.domain.model.CoffeeDomainModel
import kotlinx.coroutines.flow.Flow

class CoffeeUseCaseImpl(private val repository: CoffeeRepository) : CoffeeUseCase {
    override fun getAllHotCoffee(): Flow<List<CoffeeDomainModel>> {
        return repository.getAllHotCoffee()
    }

}